driver-core/src/test/functional/com/mongodb/client/model/expressions/DocumentExpressionsFunctionalTest.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.expressions;
+
+import org.bson.BsonDocument;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+import org.bson.types.Decimal128;
+import org.junit.jupiter.api.Test;
+
+import java.time.Instant;
+import java.util.Arrays;
+
+import static com.mongodb.client.model.expressions.Expressions.of;
+import static com.mongodb.client.model.expressions.Expressions.ofIntegerArray;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@SuppressWarnings("ConstantConditions")
+class DocumentExpressionsFunctionalTest extends AbstractExpressionsFunctionalTest {
+    // https://www.mongodb.com/docs/manual/reference/operator/aggregation/#object-expression-operators
+    // (Complete as of 6.0)
+
+    private static DocumentExpression ofDoc(final String ofDoc) {
+        return of(BsonDocument.parse(ofDoc));
+    }
+
+    private final DocumentExpression a1 = ofDoc("{a: 1}");
+    private final DocumentExpression ax1ay2 = ofDoc("{a: {x: 1, y: 2}}");
+
+    @Test
+    public void literalsTest() {
```

Comment:
Currently, `DocumentExpression` can only be obtained from a literal or as a value of a field of a different `DocumentExpression`. This means that all `DocumentExpression`s can only come from a literal (not necessary directly) with no way to have a `DocumentExpression` that represents data from a database. My understanding is that MQL "applies" `$setField` to documents that potentially come from a database via the [system variables](https://www.mongodb.com/docs/manual/reference/aggregation-variables/#system-variables) `$ROOT` and `$CURRENT`, but there appears to be no way of creating a `DocumentExpression` for either. Am I misunderstanding something, or is this work that is planned for an upcoming PR?

Reply:
There will be a `current()` shim method that I will add when I write the integration-with-existing tests.

Reply:
Let's have a separate ticket and PR for `current()`. It's important enough to deserve top billing.

Reply:
https://github.com/mongodb/mongo-java-driver/pull/1057